[RUM guide] Initialize your native SDK before RN starts (#22935)
DataDog/documentation: content/en/real_user_monitoring/guide/initialize-your-native-sdk-before-react-native-starts.md

---
title: Initialize Your Native SDK Before React Native Starts
kind: guide
description: Learn how to initialize your native SDK before React Native starts
further_reading:
- link: real_user_monitoring/reactnative/
  tag: Documentation
  text: Learn about React Native monitoring
algolia:
  tags: ['bots']
---

## Overview

By default, the React Native SDK initializes the native SDK when calling `DdSdkReactNative.initialize(config)` on the JS layer, or using the `DatadogProvider`. As a result, the SDK does not capture native crashes that occur prior to the initialization being called on the JS layer. Starting from v2.3.0, you can initialize your native SDK in order for Datadog to capture any crashes before the React Native layer starts.

## Configuration

To initialize your native SDK before React Native has started:

1. Create a `datadog-configuration.json` file at the root of the `react-native` project with the following structure:
   
   ```json
   {
     "$schema": "./node_modules/@datadog/mobile-react-native/datadog-configuration.schema.json",
     "configuration": {
     }
   }
   ```

>    The `"$schema"` attribute here helps most modern Integrated Development Environments (IDE) to show errors if the configuration is incomplete or invalid, as well as autocomplete enabled.

Reviewer: ```suggestion
   The `"$schema"` attribute here enables autocomplete and helps most modern Integrated Development Environments (IDE) to show errors if the configuration is incomplete or invalid.
```

Author: Thanks @hestonhoffman !
